qiita_db/test/test_commands.py

```diff
@@ -462,6 +465,164 @@ def test_python_patch(self):
+            "lab_person_id": StudyPerson(1)
+        }
+        self.no_ppd_study = Study.create(
+            User('test@foo.bar'), "Test study", [1], info)
+
+        # Get the directory where the preprocessed data is usually copied.
+        _, self.db_ppd_dir = get_mountpoint('preprocessed_data')[0]
+
+    def tearDown(self):
+        for fp in self._filepaths_to_remove:
+            if exists(fp):
+                remove(fp)
+        for dp in self._dirpaths_to_remove:
+            if exists(fp):
+                rmtree(dp)
+
+    def test_update_preprocessed_data_from_cmd_error_no_ppd(self):
+        with self.assertRaises(ValueError):
+            update_preprocessed_data_from_cmd(self.test_slo,
+                                              self.no_ppd_study.id)
+
+    def test_update_preprocessed_data_from_cmd_error_missing_files(self):
+        with self.assertRaises(ValueError):
+            update_preprocessed_data_from_cmd(self.missing_slo, 1)
+
+    def test_update_preprocessed_data_from_cmd(self):
+        exp_ppd = PreprocessedData(Study(1).preprocessed_data()[0])
+        exp_fps = exp_ppd.get_filepaths()
+
+        # The original paths mush exist, but they're not included in the test
+        # so create them here
+        for _, fp, _ in exp_fps:
+            with open(fp, 'w') as f:
+                f.write("")
+
+        next_fp_id = get_count('qiita.filepath') + 1
+        exp_fps.append(
+            (next_fp_id,
+             join(self.db_ppd_dir, "%s_split_library_log.txt" % exp_ppd.id),
+             'log'))
+
+        ppd = update_preprocessed_data_from_cmd(self.test_slo, 1)
+
+        # Check that the modified preprocessed data is the correct one
+        self.assertEqual(ppd.id, exp_ppd.id)
+
+        # Check that the filepaths returned are correct
+        # We need to sort the list returned from the db because the ordering
+        # on that list is based on db modification time, rather than id
+        obs_fps = sorted(ppd.get_filepaths())
+        self.assertEqual(obs_fps, exp_fps)
+
+        # Check that the checksums have been updated
+        sql = "SELECT checksum FROM qiita.filepath WHERE filepath_id=%s"
+
+        # Checksum of the fasta file
+        obs_checksum = self.conn_handler.execute_fetchone(
+            sql, (obs_fps[0][0],))[0]
+        self.assertEqual(obs_checksum, '3532748626')
+
+        # Checksum of the fastq file
+        obs_checksum = self.conn_handler.execute_fetchone(
+            sql, (obs_fps[1][0],))[0]
+        self.assertEqual(obs_checksum, '2958832064')
+
+        # Checksum of the demux file
+        # The checksum is generated dynamically, so the checksum changes
+        # We are going to test that the checksum is not the one that was
+        # before, which correspond to an empty file
+        obs_checksum = self.conn_handler.execute_fetchone(
+            sql, (obs_fps[2][0],))[0]
+        self.assertNotEqual(obs_checksum, '852952723')
```

Comment:
I don't know how likely this would be but could we be getting an empty string? Or something that is not a checksum? if likely enough, then we might want to check for length and composition of the string.